clients/cobol-lsp-vscode-extension/src/services/util/SubroutineUtils.ts

```diff
@@ -11,24 +11,26 @@
  * Contributors:
  *   Broadcom, Inc. - initial API and implementation
  */
-import { searchCopybookInExtensionFolder } from "./FSUtils";
-import { COBOL_EXT_ARRAY } from "../../constants";
+import { COBOL_EXT_ARRAY_CASE_INSENSITIVE } from "../../constants";
 import { SettingsService } from "../Settings";
-import { Uri } from "vscode";
-
+import * as vscode from "vscode";
 /**
- * This function try to resolve a given subroutine by searching COBOL source file with the same name
- * in local file system in directories specified in settings.
+ * This function try to resolve a given subroutine by searching COBOL source file
+ * with the same name in local workspace directories specified in settings.
  * @param name the name of subroutine
  * @return subroutine file URI if it was found or undefined otherwise
  */
-export function resolveSubroutineURI(storagePath: string, name: string): Uri {
-  const folders: string[] | undefined =
-    SettingsService.getSubroutineLocalPath();
-  return searchCopybookInExtensionFolder(
-    name,
-    folders,
-    COBOL_EXT_ARRAY,
-    storagePath,
-  )!;
+export async function resolveSubroutineURI(name: string) {
+  const subroutinePaths = SettingsService.getSubroutineLocalPath();
+
+  if (subroutinePaths && subroutinePaths.length > 0) {
+    const pattern = `{${subroutinePaths.join(",")}}/**/${name}{${COBOL_EXT_ARRAY_CASE_INSENSITIVE.join(",")}}`;
+    const uris = await vscode.workspace.findFiles(pattern, null, 1);
```

Comment:
Can these be also absolute paths? Does it work with findFiles?

Reply:
In the extension readme, the relative paths are recommended. But `findFiles` can work with absolute paths as well, the path just needs to be passed in as `RelativePattern`, so I added that.
I don't like that we need to use `path` module to determine if the path is absolute, but I haven't found any VSCode API to do that.